scripts/add-chain.sh

```diff
@@ -78,19 +78,56 @@ genesis:
 EOF
 
 
-# add extra addresses data
+# infer appropriate L1_RPC_URL
+case $SUPERCHAIN_TARGET in
+    "mainnet")
+        L1_RPC_URL="https://ethereum-mainnet-rpc.allthatnode.com"
+        ;;
+    "sepolia")
+        L1_RPC_URL="https://ethereum-sepolia-rpc.allthatnode.com"
+        ;;
+    *)
+        echo "Unsupported Superchain Target $SUPERCHAIN_TARGET"
+        exit 1
+        ;;
+esac
+
+# scrape addresses from static deployment artifacts
+AddressManager=$(jq -j .address $DEPLOYMENTS_DIR/AddressManager.json)
+L1CrossDomainMessengerProxy=$(jq -j .address $DEPLOYMENTS_DIR/L1CrossDomainMessengerProxy.json)
+L1ERC721BridgeProxy=$(jq -j .address $DEPLOYMENTS_DIR/L1ERC721BridgeProxy.json)
+L1StandardBridgeProxy=$(jq -j .address $DEPLOYMENTS_DIR/L1StandardBridgeProxy.json)
+L2OutputOracleProxy=$(jq -j .address $DEPLOYMENTS_DIR/L2OutputOracleProxy.json)
+OptimismMintableERC20FactoryProxy=$(jq -j .address $DEPLOYMENTS_DIR/OptimismMintableERC20FactoryProxy.json)
+SystemConfigProxy=$(jq -j .address $DEPLOYMENTS_DIR/SystemConfigProxy.json)
+OptimismPortalProxy=$(jq -j .address $DEPLOYMENTS_DIR/OptimismPortalProxy.json)
+SystemConfigProxy=$(jq -j .address $DEPLOYMENTS_DIR/SystemConfigProxy.json)
+
+# scrape remaining address live from the chain
+SuperchainConfig=$(cast call $OptimismPortalProxy "superchainConfig()(address)" -r $L1_RPC_URL) || "" # first command could fail if bedrock
+Guardian=$(cast call $SuperchainConfig "guardian()(address)" -r $L1_RPC_URL) || $(cast call $OptimismPortalProxy "GUARDIAN()(address)" -r $L1_RPC_URL) # first command could fail if bedrock
```

Comment:
Another +1 point for transcribing into a different language: the error handling here _should_ work, but it is not pretty.